File: 1440-convert-integer-to-the-sum-of-two-no-zero-integers/convert-integer-to-the-sum-of-two-no-zero-integers.py
class Solution:
    def getNoZeroIntegers(self, n: int) -> List[int]:
        for i in range(1,n):
            a = n - i
            b = i
            m = list(str(a))
            l = list(str(b))
            if ('0' not in m) and ('0' not in l):
                return [a,b]

        return [] 

File: 1440-convert-integer-to-the-sum-of-two-no-zero-integers/convert-integer-to-the-sum-of-two-no-zero-integers.java
class Solution {
    public static Boolean safe(int a){
        while(a > 0){
            int r = a % 10;
            if(r == 0) return false;
            a /= 10;
        }

        return true;
    }
    public int[] getNoZeroIntegers(int n) {
        int[] ans = new int[2];

        for(int i=1;i<n;i++){
            int a = n - i;
            int b = i;

            if(safe(a) && safe(b)){
                ans[0] = a;
                ans[1] = b;
                break;
            }
        }

        return ans;
    }
}